### PR TITLE
update the transformer section with query parameter etc.

### PR DESCRIPTION
**Pull request by kwangsuk**

### CODE REVIEW

File: doc/mgmt/Developer Guide.md [hunk 1]
```diff
@@ -655,404 +646,1854 @@ $ tools/xfmr/annotate.sh
...
+Query parameters is supported only for GET request from NBI. The QPs supported by mgmt-framework are depth, content and fields.
+Please refer to the HLD for more details on QP support in mgmt-framework.
+https://github.com/project-arlo/SONiC/blob/master/doc/mgmt/SONiC_QueryParameterForGet_HLD.md
 
-// XfmrParams contains context, data passed to the callback functions.
+###### 2.5.8.1 Query parameters in Transformer
+
+"QueryParams" struct will be used by the transformer infra to process the Query parameters and will be shared with the app subtree. This will part of "XfmrParams" struct.
+
+The data structures as are give below.
```
Comment (jeffyin-dell) [inline doc/mgmt/Developer Guide.md]: Should say "The data structures are as below:"

File: doc/mgmt/Developer Guide.md [hunk 2]
```diff
@@ -655,404 +646,1854 @@ $ tools/xfmr/annotate.sh
...
-    uri string
 
-    // ygRoot is the ygot struct initialized by translib for the requestUri. -- (RW)
-    ygRoot *ygot.GoStruct
+###### 2.5.8.2 Example to support query parameters
 
-    // oper indicates the current operation requested by translib. -- (RO)
-    // 1=GET, 2=CREATE, 3=REPLACE, 4=UPDATE, 5=DELETE
-    oper int
+Example implementation of subtree callbacks using query parameters can be seen below. Note that the data trees not annotated with subtree callbacks are implicitly handled by transformer core, that it does not require codes to support query parameters.
```
Comment (jeffyin-dell) [inline doc/mgmt/Developer Guide.md]: That last clause should be "and they do not require additional code to support query parameters."

File: doc/mgmt/Developer Guide.md [hunk 3]
```diff
@@ -655,404 +646,1854 @@ $ tools/xfmr/annotate.sh
...
 
-    // dbDataMap is the internal map referenced by callback. -- (RO)
-    dbDataMap *map[db.DBNum]map[string]map[string]db.Value
+var DbToYang_bgp_routes_get_xfmr SubTreeXfmrDbToYang = func(inParams XfmrParams) error {
 
-    // subOpDataMap is a map[OPCODE]map[DB_NUM]map[TABLE]map[KEY]db.Value -- (RW)
-    // It should be filled by YangToDb_xxx callbacks to record sub-operations.
-    subOpDataMap map[int]*RedisDbMap
+        var err error
+        oper_err := errors.New("Opertational error")
```
Comment (jeffyin-dell) [inline doc/mgmt/Developer Guide.md]: "Operational"

**Review by jeffyin-dell — COMMENTED** [review]
I see a lot of changes outside of just the Query Parameters section. Can you be more specific about what else was updated? Where did the updates for the other sections come from, and has anything major been changed?

**Comment by kwangsuk** [comment]
> I see a lot of changes outside of just the Query Parameters section. Can you be more specific about what else was updated? Where did the updates for the other sections come from, and has anything major been changed?

Some sections are updated with more explanation & clarification where needed, also added the best practices

**Pull request merged**